### ISSUE TITLE
Problem with decorator in CRUD

### ISSUE THREAD

**Issue by karakal** [atk4/ui#1079]
I have a model with a few classed with fields that have the type date.

```
class Fahrer extends WEModel
{
    public $table = 'fahrer';
    public $id_field = 'id';
    public $caption = 'Fahrer';
    public $title_field = 'Person';

    function init()
    {
        parent::init();
        $this->addField('ablaufdatum',['caption'=>'Ablaufdatum','type'=>'date']);
...
```

When I use that class in a CRUD and I try to add a decorator it ends in a really awful exception:
![grafik](https://user-images.githubusercontent.com/2448293/78801460-06516980-79bd-11ea-923f-da5ebb262c0e.png)

Even if the decorator is used on another field:
```
$crud->addDecorator('Person',['Link', 'befaehigungen.php?fahrer_id={$id}']);
```

When I don't use the decorator everything is fine..

The problem is new for atk4/ui:2.1.0-alpha1 with atk4/core:2.0.2

**Comment by DarkSide666** [comment]
Issue: Model field value should be cast to string.
Possible solution: use Field->toString() method before setting value of template tag.

**Comment by karakal** [comment]
Sorry, what should I do?

**Comment by PhilippGrashoff** [comment]
nothing at the moment it seems. There might be a bug that instead of string the field object is passed to Template->set() which causes this exception. The problem is in TableColumn/Link.php in line 149.

**Comment by karakal** [comment]
I had a look at it again because all the Mastercrud Views were working just fine. So I looked into the Sourcecode of Mastercrud.

If you change
`$crud->addDecorator('Person',['Link', 'befaehigungen.php?fahrer_id={$id}']);`
to
`$crud->addDecorator('Person', ['Link', ['befaehigungen'], ['fahrer_id'=>'id']]);`
it is working again...

**Comment by DarkSide666** [comment]
Ah yes, but how is that related to `type=date` ?

**Comment by karakal** [comment]
How should I know? I just found the bug, provided you with the information I could gather from my application.

If I have in my class a field with the type date and use the decorator like I used to use it before, I get an exception. When I remove all instances of fields with type date, the decorator works again.

When I use the decorator in the second way, I can have fields with type date in my model-class.

Somehow the line 149 in the function getHTMLTags($row,$field) in the TableColumn/Link.php doesn't like the first way of declaration of the Decorator when fields with type date are present in the model.

And this is the same in all the instances where I use the decorator. So two possibilities:

- I use(d) the decorator wrong. But then there must have be any change because it worked just fine before.
- There is somehow somewhere a bug.
